src/component/navbar.rs

```diff
@@ -0,0 +1,303 @@
+                }
+
+                match props.disabled {
+                    true => {
+                        html! {
+                            <li class="nav-item">
+                                <a class={classes} tabindex="-1" aria-disabled="true" href={props.url.clone()}>
+                                    {props.text.clone()}
+                                </a>
+                            </li>
+                        }
+                    },
+                    false => {
+                        html! {
+                            <li class="nav-item">
+                                <a class={classes} href={props.url.clone()}>
+                                    {props.text.clone()}
+                                </a>
+                            </li>
+                        }
+                    }
+                }
+            },
+            false => {
+                html! {
+                    <NavDropdown text={props.text.clone()} id={props.id.clone()}>
+                        { for props.children.iter() }
+                    </NavDropdown>
+                }                
+            }
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq)]
+pub enum BrandType {
+    BrandSimple { text: String, url: Option<String> },
+    BrandImage { 
+        /// browser-accessible url to the brand image
+        image_url: String, 
+        /// descriptive text for screen reader users
+        alt: String, 
+        dimension: Option<Dimension>
+    },
+    BrandCombined {
+        text: String, 
+        /// hyperlink destination for brand text
+        url: Option<String>,
+        /// browser-accessible url to the brand image
+        image_url: String, 
+        /// descriptive text for screen reader users
+        alt: String, 
+        dimension: Option<Dimension>
+    }
+}
+
+pub struct NavBar { }
+
+#[derive(Properties, Clone, PartialEq)]
+pub struct NavBarProps {
+    #[prop_or_default]
+    pub children: Children,
+    #[prop_or_default]
+    pub class: String,
+
+    /// the id of the div that contains the nav-items
+    #[prop_or_default]
+    pub nav_id: String,
+
+    #[prop_or_default]
+    pub expanded: bool,
+
+    #[prop_or_default]
+    pub brand: Option<BrandType>
+}
+
+impl Component for NavBar {
+    type Message = ();
+    type Properties = NavBarProps;
+
+    fn create(_ctx: &Context<Self>) -> Self {
+        Self {}
+    }
+
+    fn view(&self, ctx: &Context<Self>) -> Html {
+        let props = ctx.props();
+
+        let expanded = String::from(match &props.expanded {
+            true => {
+                "true"
+            },
+            false => {
+                "false"
+            }
+        });
+
+        let mut classes = Classes::new();
+        classes.push("navbar");
+        classes.push(props.class.clone());
+
+        let brand = match &props.brand {
+            None => html!{},
+            Some(b) => {
+                match b {
```

Comment:
I'm not very happy with the depth of this nesting. Extracting to a function may clean this up nicely, but it'll never be reused.